translate.go

```diff
@@ -0,0 +1,119 @@
+package i18n
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/jibber_jabber"
+)
+
+var (
+	localeRegexp  string          = "^[a-z]{2}([_-][A-Z]{2}){0,1}$"
+	log                           = golog.LoggerFor("i18n")
+	fs            http.FileSystem = http.Dir("locale")
+	defaultLocale string          = "en_US"
+	defaultLang   string          = "en"
+	trMutex       sync.RWMutex
+	trMap         map[string]string
+)
+
+// T translates the given key into a message based on the current locale,
+// formatting the string using the supplied (optional) args. This method will
+// fall back to other locales if the key isn't defined for the current locale.
+// The search order (with examples) is as follows:
+//
+//   1. current locale        (zh_CN)
+//   2. lang only             (zh)
+//   3. default locale        (en_US)
+//   4. lang only of default  (en)
+//
+func T(key string, args ...interface{}) string {
+	trMutex.RLock()
+	defer trMutex.RUnlock()
+	s := trMap[key]
```

Comment:
It will not actually , though it's bad practice. See https://blog.golang.org/go-maps-in-action#TOC_2. As it's useless to initialise it or check each time, add comment and test coverage would be ok.

>  A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic; don't do that.